relengapi/blueprints/tooltool/grooming.py

```diff
@@ -0,0 +1,168 @@
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+import hashlib
+import logging
+import sqlalchemy as sa
+
+from datetime import timedelta
+from flask import current_app
+from relengapi.blueprints.tooltool import tables
+from relengapi.blueprints.tooltool import util
+from relengapi.lib import badpenny
+from relengapi.lib import celery
+from relengapi.lib import time
+
+log = logging.getLogger(__name__)
+
+
+@badpenny.periodic_task(seconds=600)
+def check_pending_uploads(job_status):
+    """Check for any pending uploads and verify them if found."""
+    session = current_app.db.session('tooltool')
+    for pu in tables.PendingUpload.query.all():
+        check_pending_upload(session, pu)
+    session.commit()
+
+
+@badpenny.periodic_task(seconds=3600)
+def replicate(job_status):
+    """Replicate objects between regions as necessary"""
+    # fetch all files with at least one instance, but not a full complement
+    # of instances
```

Comment:
AWS has announced Cross-Region Replication: https://docs.aws.amazon.com/AmazonS3/latest/dev/crr.html. Probably, could use it instead.